Ri mi start event

krisiye opened the pull request:
Updates to reflect the following:

Moved Quiz to the object - ID.
Added extensions to represent - some quiz attributes. (Assuming there is no real home for these RI specific attributes in the spec)

krisiye committed:
Adding extensions and updates to object representing quiz

krisiye committed:
Moving to RI specific dir, since MI may look different

krisiye committed:
Updates to membership. Drops section references. Fixes to other typos

krisiye committed:
fix typo on IRI

odowdj commented on src/test/resources/fixtures/hmh-ri/caliperEventAssessmentStarted.json:
```diff
@@ -0,0 +1,45 @@
+         "Learner"
+      ]
+   },
+   "extensions": {
```
I'm not sure if these specific extensions should reside out at the same level as Actor and the other core entities etc.... should they be under some inner caliper entity such as target or object maybe....?

Does having them out at the same level as Actor and the other core Caliper entities mean you lose context on what they are specifically associated with?

krisiye committed:
Updates to AssessmentStarted. Added AssessmentCompleted. Includes additional quiz attributes, extensions attached to the Assessment Object instead of root, IRI's for quiz, user, org etc.

krisiye commented:
1. Added additional quiz attributes (quiz state, grade, assessment description, and recommendations)
   - quiz state - The quiz state for the RI quiz
   - grade - the quiz grade applied
   - assessment description - indicative of the source of the start Lexile. 
   - Recommendations will show up only for a completed event and will be used for reporting.
2. Added fixtures for the Completed Event
  - Includes Recommendations (optional)
3. Added IRI’s where appropriate

krisiye commented:
Closing this PR - in favor of suggestions from @odowdj and @padraig-ohiceadha  to use AssessmentEvent, AssessmentItemEvent and GradeEvent.
